kb/deadline-exceeded-troubleshooting.md

```diff
@@ -0,0 +1,117 @@
+---
+slug: deadline-exceeded-troubleshooting
+title: DeadlineExceeded error troubleshooting
+tags:
+  - kb-article
+date: 2022-11-09T00:00:00Z
+---
+
+All client-side requests made to the [Temporal Cluster](/concepts/what-is-a-temporal-cluster) are [gRPC requests](https://grpc.io/docs/what-is-grpc/core-concepts/#deadlines).
+Sometimes, when these frontend requests can't be completed, you'll see this particular error message: `Context: deadline exceeded`.
+Network interruptions, timeouts, server overload, and Query errors are some of the causes of this error.
+
+The following sections discuss the nature of this error and how to troubleshoot it.
+
+<!-- truncate -->
+
+#### Check Frontend Service logs
+
+:::note
+
+Cloud users cannot access some of the logs needed to diagnose the source of the error.
+
+If you're using Temporal Cloud, check your Workflow History for error messages and create a [support ticket](https://support.temporal.io/) with Temporal.
```

Comment:
What does the workflow history have to do with deadline exceeded?

Reply:
if your client query calls for example are failing with deadline exceeded
you could check wf history to see if you have a local activity thats extending workflow task that could be a good indication of why query might be failing..one thing i know at least :)

Reply:
another situation could be a workflow thats stuck
for example due to non-deterministic error
in that case wht workflow task keeps replaying and the query cannot complete, think history can help there too

Reply:
This should be clarified in the doc.

The only correlation I know of between workflow history and context deadline exceeded would be on query time outs and even then it's not guaranteed.